cpp/src/arrow/extension/CMakeLists.txt

```diff
@@ -15,10 +15,16 @@
 # specific language governing permissions and limitations
 # under the License.
 
+set(CANONICAL_EXTENSION_TESTS uuid_test.cc)
+
+if(ARROW_JSON)
```

Comment:
this shouldn't be required now that you are building extensions even without `ARROW_JSON`, right?

Reply:
See my reply below.